### PR TITLE
feat(core): tree-sitter structural outlines for read_file

### PR DESCRIPTION
**Pull request by chaliy**
## What
When `read_file` returns a truncated result, the response now includes structural outlines of the unread file portions — function/class/method signatures extracted via tree-sitter.

## Why
Copilot's most powerful context protection pattern: agents see file structure even when reading a narrow range. No other open-source agent platform has replicated this. With a single partial read, the agent can orient itself in any file — knowing what functions, structs, and classes exist in the unread portions.

## How
- New `crates/core/src/outline.rs` module with tree-sitter parsing
- Phase 1 language support: Rust, TypeScript/JavaScript, Python
- Plain-text outlines appended to `content` when truncated (simplest approach, matches Copilot UX)
- Behind `tree-sitter-outlines` feature flag, enabled in server and worker
- Graceful fallback: unsupported languages return no outline
- 8 tests covering all 3 languages, formatting, and edge cases
- Updated docs in `docs/advanced/read-tools.mdx` (replaced Coming Soon with actual docs)

### Outline format
```
--- Outline of lines 101-500 (not shown) ---
// L105: fn process(input: &Input) -> Result<Output> { ... }
// L180: fn validate(output: &Output) -> bool { ... }
// L220: struct Config { ... }
// L235: impl Config { ... }
  // L236: fn load(path: &str) -> Result<Self> { ... }
```

## Risk
- **Medium** — adds new dependencies (tree-sitter + 3 language grammars)
- Feature-flagged so can be disabled without code changes
- Parsing only happens when content is truncated (no overhead for small files)
- tree-sitter parsing is fast (~1ms for typical source files)

## Security
- **TM-TOOL**: Outline generated from file content already in memory — no new file I/O or paths.
- **TM-DOS**: Parsing bounded by existing content. `READ_FILE_HARD_BYTE_CAP` (50 KB) still applies.
- **Dependency risk**: tree-sitter crates are from the official tree-sitter org, widely used. Behind feature flag.
- No auth, authz, tenant, SQL, or prompt injection changes.

## Checklist
- [x] Tests added or updated (8 new tests for outline module)
- [x] Backward compatibility considered (feature-flagged, graceful fallback)
- [x] Security review performed against relevant threat model categories
- [x] All review comments addressed

Fixes EVE-248

### CODE REVIEW

File: crates/core/src/outline.rs [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+    }
+}
+
+#[cfg(feature = "tree-sitter-outlines")]
+fn detect_language(path: &str) -> Option<tree_sitter::Language> {
+    let lower = path.to_lowercase();
+    if lower.ends_with(".rs") {
+        Some(tree_sitter_rust::LANGUAGE.into())
+    } else if lower.ends_with(".ts") || lower.ends_with(".mts") {
+        Some(tree_sitter_typescript::LANGUAGE_TYPESCRIPT.into())
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/outline.rs]: Fixed: unified language_id() to return Option<LangId> instead of falling back to Rust. Added .mts support. Unknown extensions now return None, skipping outline generation entirely.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/core/src/outline.rs [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+                        .map(|n| node_text(n, source).to_string())
+                        .unwrap_or_default();
+                    let sig = ts_fn_signature(value, source);
+                    items.push(OutlineItem {
+                        kind: "function",
+                        name,
+                        signature: sig,
+                        line: child.start_position().row + 1,
+                        children: vec![],
+                    });
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/outline.rs:466]: Fixed: for lexical declarations with arrow functions, the variable name is now prepended to the signature (e.g., `const foo = (x: string) =>`). Check detects signatures starting with `(` or `async (` and prefixes with `const {name} = `.

---
_Generated by [Claude Code](https://claude.ai/code)_
